lib/Connection.php

```diff
@@ -249,7 +256,7 @@ private function closeSocket() {
         $this->outputBuffer = "";
         $this->outputBufferLength = 0;
 
-        if (is_resource($this->socket)) {
+        if (!$this->persistent && is_resource($this->socket)) {
```

Comment:
The `fclose()` should be executed on explicit `close()` call, but not when called via `__destruct()`, if persistent.